fe/fe-core/src/main/java/org/apache/doris/analysis/BulkLoadDataDesc.java

```diff
@@ -0,0 +1,376 @@
+    }
+
+    public List<String> getColumnsFromPath() {
+        return columnsFromPath;
+    }
+
+    public Map<String, String> getProperties() {
+        return dataProperties;
+    }
+
+    // Change all the columns name to lower case, because Doris column is case-insensitive.
+    private List<String> columnsNameToLowerCase(List<String> columns) {
+        if (columns == null || columns.isEmpty() || "json".equals(this.formatDesc.fileFormat)) {
+            return columns;
+        }
+        List<String> lowerCaseColumns = new ArrayList<>();
+        for (int i = 0; i < columns.size(); i++) {
+            String column = columns.get(i);
+            lowerCaseColumns.add(i, column.toLowerCase());
+        }
+        return lowerCaseColumns;
+    }
+
+    public String toInsertSql(Map<String, String> properties) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("INSERT INTO ");
+        sb.append("`").append(dbName).append("`");
+        sb.append("`").append(tableName).append("`");
+        sb.append("(");
+
+        List<String> targetColumns = getTargetColumnsFromTable(dbName, tableName);
+        verifyLoadColumns(fileFieldNames, targetColumns);
+
+        List<String> mappingTargetColumns = tryToTransFormToMappingColumns(targetColumns);
+        appendColumnFromPathColumns(mappingTargetColumns);
+        appendVirtualColumnForDeleteAndSeq(mappingTargetColumns);
+        Joiner.on(", ").appendTo(sb, mappingTargetColumns);
+        sb.append(")");
+
+        sb.append(" SELECT ");
+        List<String> mappingSrcColumns = tryToSetMappingColumns(fileFieldNames);
+        appendColumnValuesFromPathColumns(mappingSrcColumns);
+        appendDeleteOnConditions(mappingSrcColumns);
+        Joiner.on(", ").appendTo(sb, mappingSrcColumns);
+        sb.append(" FROM ");
+
+        // TODO: check if s3
+        // if is s3 data desc type
+        sb.append(" s3( ");
+        Iterator<Map.Entry<String, String>> iter = properties.entrySet().iterator();
+        while (iter.hasNext()) {
+            Map.Entry<String, String> entry = iter.next();
+            sb.append("\"");
+            sb.append(entry.getKey());
+            sb.append("\"=\"");
+            sb.append(entry.getValue());
+            sb.append("\"");
+            if (iter.hasNext()) {
+                sb.append(",");
+            }
+        }
+        // properties.isEmpty();
+        // append s3 properties
+        sb.append(")");
+        Expression rewrittenWhere = rewriteByColumnMappingSet(whereExpr);
+        String rewrittenWhereClause = rewriteByPrecedingFilter(rewrittenWhere, precedingFilterExpr);
+        sb.append(" WHERE ").append(rewrittenWhereClause);
+        return sb.toString();
+    }
+
+    private void verifyLoadColumns(List<String> fileFieldNames, List<String> targetColumns) {
+    }
+
+    private String rewriteByPrecedingFilter(Expression whereExpr, Expression precedingFilterExpr) {
+        return whereExpr.toSql();
```

Comment:
I am not sure that `toSql()` to recover the origin sql string exactly.
How about just save the origin sql string when parsing the Load Stmt?